kentik_api_library/kentik_api/api_connection/api_connector.py

```diff
@@ -1,32 +1,50 @@
 # Standard library imports
 import logging
+from http import HTTPStatus
 from typing import Optional, Dict, Any
 
 # Third party imports
 import requests
 
 # Local application imports
-from kentik_api.auth.auth import KentikAuth
 from kentik_api.api_calls.api_call import APICall, APICallMethods
 from kentik_api.api_connection.api_call_response import APICallResponse
+from kentik_api.auth.auth import KentikAuth
+from kentik_api.public.errors import (
+    AuthError,
+    BadRequestError,
+    NotFoundError,
+    RateLimitExceededError,
+    UnavailabilityError,
+    KentikAPIError,
+)
 
 
 class APIConnector:
     """ APIConnector implements APIConnectorProtocol. Allows sending authorized http requests to Kentik API """
 
     DEFAULT_HEADERS = {"Content-Type": "application/json"}
-    BASE_API_US_URL = "https://api.kentik.com/api"
-    BASE_API_EU_URL = "https://api.kentik.eu/api"
 
     def __init__(self, api_url: str, auth_email: str, auth_token: str) -> None:
         self._api_url = api_url
         self._auth = KentikAuth(auth_email, auth_token)
         self._logger = logging.getLogger(__name__)
 
     def send(self, api_call: APICall, payload: Optional[Dict[str, Any]] = None) -> APICallResponse:
-        url = self._get_api_query_url(api_call.url_path)
-        response: requests.Response
+        try:
+            response = self._do_request(api_call, payload)
+        except requests.Timeout as e:
```

Comment:
this Timeout seems to be subject to retrying the request. 
Would be nice to prepare a hierarchy of retry-able exceptions that could be intercepted and handled by retrying mechanism, eg TryAgainError as a root and children TimedOut and RateLimitExceeded, there is more http error codes that could translate to TryAgainError but this is good start

Reply:
Right now there are 2 errors that are retryable:
- RateLimitExceededError raised on HTTP 429
- UnavailabilityError which is raised on rest of retryable errors: requests.Timeout (ConnectTimeout, ReadTimeout) and HTTP 503, 504

Client code can intercept both of them for retrying mechanism. It just needs to catch 2 exceptions.
1. I could make `RateLimitExceededError` and `UnavailabilityError` to inherit from `TryAgainError`, but I doubt it gives much convenience for catching errors.
2. I could make errors more specific and have e.g. RateLimitExceeded, Timeout, ServiceUnavaliable, GatewayTimeout, but I am not sure it is useful for the user.

For both improvements I am in doubt so I kept the solution with minimal API surface.

Reply:
The benefit from having exception hierarchy for retrying is that if we later realize that there is more http error codes that are candidates for retrying, we just add the additional ones to the hierarchy and publish new version of library, and the clients who use the lib will instantaneously benefit from it.
This is not vital, but handy

The ConnectionTimeout and ReadTimeout is good finding; we should explicitly configure our http client in APIConnector to timeout after specified period of time

Reply:
I added IntermittentError as base error for retryable errrors.